File: src/main/java/ch/study/Weather.java
package ch.study;

public class Weather {

    private final String description;
    private Color color;

    public Weather(int degrees) {
        String description;
        Color color;
        if (degrees < 10) {
            description = "cold";
            color = Color.BLUE;
        } else if (degrees < 25) {
            description = "mild";
            color = Color.ORANGE;
        } else {
            description = "hot";
            color = Color.RED;
        }
        this.description = description;
        this.color = color;
    }

    public enum Color {
        BLUE,
        ORANGE,
        RED;
    }
}

File: src/main/java/ch/study/Weather.kt
package ch.study

class Weather(degrees: Int) {
    private val description: String
    private val color: Color

    init {
        val (description, color) = when {
            degrees < 10 -> "cold" to Color.BLUE
            degrees < 25 -> "mild" to Color.ORANGE
            else -> "hot" to Color.RED
        }
        this.description = description
        this.color = color
    }

    enum class Color {
        BLUE, ORANGE, RED
    }
}